spec/integration/discourse_prometheus_alert_receiver/receiver_controller_spec.rb

```diff
@@ -930,20 +930,34 @@
 
         let(:topic) { Topic.find_by(id: receiver["topic_map"][alert_name]) }
 
-        before do
-          assignee_group.add(bob)
+        before { assignee_group.add(bob) }
+
+        it "creates a new topic with the right assignment when `commonAnnotations.topic_assignee` field is present in the payload" do
           payload["commonAnnotations"]["topic_assignee"] = "bobtheangryflower"
-        end
 
-        it "creates a new topic" do
           expect do post "/prometheus/receiver/#{token}", params: payload end.to change {
             Topic.count
           }.by(1)
 
           expect(topic.assigned_to).to eq(bob)
         end
 
-        describe "when prometheus_alert_receiver_enable_assign is false" do
+        it "creates a new topic with the right assignment when `commonAnnotations.topic_group_assignee` field is present in the payload" do
+          assignee_group.update!(assignable_level: Group::ALIAS_LEVELS[:only_admins])
+
+          payload["commonAnnotations"]["topic_group_assignee"] = assignee_group.name
+
+          expect do post "/prometheus/receiver/#{token}", params: payload end.to change {
+            Topic.count
+          }.by(1)
+
+          expect(response.status).to eq(200)
+          expect(topic.assigned_to).to eq(assignee_group)
+
+          topic.destroy!
```

Comment:
What is the destroy for?

Reply:
Hmm this was from old code and I don't know why it is there too. Have removed it.